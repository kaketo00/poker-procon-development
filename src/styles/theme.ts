import { extendTheme } from '@chakra-ui/react';
import { Zen_Kaku_Gothic_New } from 'next/font/google';

const nextFont = Zen_Kaku_Gothic_New({
  weight: ['400', '700', '900'],
  subsets: ['latin'],
});

const theme = extendTheme({
  fonts: {
    heading: nextFont.style.fontFamily,
    body: nextFont.style.fontFamily,
  },
  colors: {
    brand100: {
      100: '#0DD203', // green 100%
      200: '#005CFF', // blue 100%
      300: '#B4171E', // red 100%
      600: '#777777', // dark-gray 100%
      700: '#D4D4D4', // gray 100%
      800: '#ECECEC', // light-gray 100%
    },
    brand80: {
      100: 'rgba(13, 210, 3, 0.8)', // green 80%
      200: 'rgba(0, 92, 255, 0.8)', // blue 80%
      300: 'rgba(180, 23, 30, 0.8)', // red 80%
      400: 'rgba(255, 255, 255, 0.8)', // white 80%
      500: 'rgba(0, 0, 0, 0.8)', // black 80%
      600: 'rgba(119, 119, 119, 0.8)', // gray 80%
    },
  },
  textStyles: {
    normal: {
      xs: {
        fontSize: '0.625rem',
        fontWeight: 'normal',
        lineHeight: '0.75rem',
      },
      sm: {
        fontSize: '0.75rem',
        fontWeight: 'normal',
        lineHeight: '1rem',
      },
      md: {
        fontSize: '1rem',
        fontWeight: 'normal',
        lineHeight: '1.25rem',
      },
      lg: {
        fontSize: '1.25rem',
        fontWeight: 'normal',
        lineHeight: '1.5rem',
      },
      xl: {
        fontSize: '1.5rem',
        fontWeight: 'normal',
        lineHeight: '1.75rem',
      },
    },
    bold: {
      xs: {
        fontSize: '0.625rem',
        fontWeight: 'bold',
        lineHeight: '0.75rem',
      },
      sm: {
        fontSize: '0.75rem',
        fontWeight: 'bold',
        lineHeight: '1rem',
      },
      md: {
        fontSize: '1rem',
        fontWeight: 'bold',
        lineHeight: '1.25rem',
      },
      lg: {
        fontSize: '1.25rem',
        fontWeight: 'bold',
        lineHeight: '1.5rem',
      },
      xl: {
        fontSize: '1.5rem',
        fontWeight: 'bold',
        lineHeight: '1.75rem',
      },
    },
    strong: {
      xs: {
        fontSize: '0.6875rem',
        fontWeight: 'black',
        lineHeight: '1rem',
      },
      sm: {
        fontSize: '1.25rem',
        fontWeight: 'black',
        lineHeight: '1.5rem',
      },
      md: {
        fontSize: '1.5rem',
        fontWeight: 'black',
        lineHeight: '2rem',
      },
      lg: {
        fontSize: '1.875rem',
        fontWeight: 'black',
        lineHeight: '2.375rem',
      },
      xl: {
        fontSize: '2.25rem',
        fontWeight: 'black',
        lineHeight: '2.75rem',
      },
    },
    heading: {
      h6: {
        fontSize: '1rem',
        fontWeight: 'black',
        lineHeight: '1.25rem',
      },
      h5: {
        fontSize: '1.25rem',
        fontWeight: 'black',
        lineHeight: '1.5rem',
      },
      h4: {
        fontSize: '1.5rem',
        fontWeight: 'black',
        lineHeight: '1.875rem',
      },
      h3: {
        fontSize: '1.75rem',
        fontWeight: 'black',
        lineHeight: '2.125rem',
      },
      h2: {
        fontSize: '2.25rem',
        fontWeight: 'black',
        lineHeight: '2.75rem',
      },
      h1: {
        fontSize: '1.875rem',
        fontWeight: 'black',
        lineHeight: '2.25rem',
      },
    },
    shadows: {
      light5: {
        100: '0px 0px 5px rgba(13, 210, 3, 0.8)',
        200: '0px 0px 5px rgba(0, 92, 255, 0.8)',
        300: '0px 0px 5px rgba(180, 23, 30, 0.8)',
        400: '0px 0px 5px rgba(255, 255, 255, 0.8)',
      },
      light10: {
        100: '0px 0px 10px rgba(13, 210, 3, 0.8)',
        200: '0px 0px 10px rgba(0, 92, 255, 0.8)',
        300: '0px 0px 10px rgba(180, 23, 30, 0.8)',
        400: '0px 0px 10px rgba(255, 255, 255, 0.8)',
      },
    },
  },
  breakpoints: {
    base: '0px', // 0 - 1279px game-template(S)
    sm: '80em', // 1280px - 1439px game-template(S)
    md: '90em', // 1440px - 1919px game-template(M)
    lg: '120em', // 1920px - 2559px game-template(M)
    xl: '160em', // 2560px - âˆž game-template(L)
    '2xl': '160em',
  },
});

export default theme;
