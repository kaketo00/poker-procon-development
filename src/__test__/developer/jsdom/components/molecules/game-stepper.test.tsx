import { ChakraProvider } from '@chakra-ui/react';
import { fireEvent, render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import type { GameStepperProps } from '@/components/molecules/game-stepper';
import GameStepper from '@/components/molecules/game-stepper';
import theme from '@/styles/theme';

const Component = (props: GameStepperProps) => (
  <ChakraProvider theme={theme}>
    <GameStepper {...props} />
  </ChakraProvider>
);

describe('GameStepper', () => {
  test('スナップショット/1ページ目', () => {
    const { asFragment } = render(
      <Component current={1} max={10} onChangeStep={jest.fn()} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/最終ページ', () => {
    const { asFragment } = render(
      <Component current={10} max={10} onChangeStep={jest.fn()} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('Prevボタンをクリックすると1ページ前に戻る', async () => {
    const onChangeStep = jest.fn();
    const { getByRole } = render(
      <Component current={2} max={10} onChangeStep={onChangeStep} />
    );

    await userEvent.click(getByRole('button', { name: 'Prev' }));

    await waitFor(() => {
      expect(onChangeStep).toHaveBeenCalledWith(1);
    });
  });

  test('Nextボタンをクリックすると1ページ次に進む', async () => {
    const onChangeStep = jest.fn();
    const { getByRole } = render(
      <Component current={1} max={10} onChangeStep={onChangeStep} />
    );

    await userEvent.click(getByRole('button', { name: 'Next' }));

    await waitFor(() => {
      expect(onChangeStep).toHaveBeenCalledWith(2);
    });
  });

  test('現在のページに数値を入力すると該当のページに移動する', async () => {
    const onChangeStep = jest.fn();
    const { getByRole } = render(
      <Component current={1} max={10} onChangeStep={onChangeStep} />
    );

    fireEvent.input(getByRole('spinbutton', { name: 'pager' }), {
      target: {
        value: '5',
      },
    });

    await waitFor(() => {
      expect(onChangeStep).toHaveBeenCalledWith(5);
    });
  });

  test('現在のページに文字を入力してもページは移動しない', async () => {
    const onChangeStep = jest.fn();
    const { getByRole } = render(
      <Component current={1} max={10} onChangeStep={onChangeStep} />
    );

    fireEvent.input(getByRole('spinbutton', { name: 'pager' }), {
      target: {
        value: '-',
      },
    });

    await waitFor(() => {
      expect(onChangeStep).not.toHaveBeenCalled();
    });
  });

  test('現在のページに最大ページ以上の値を入力してもページは移動しない', async () => {
    const onChangeStep = jest.fn();
    const { getByRole } = render(
      <Component current={1} max={10} onChangeStep={onChangeStep} />
    );

    fireEvent.input(getByRole('spinbutton', { name: 'pager' }), {
      target: {
        value: '11',
      },
    });

    await waitFor(() => {
      expect(onChangeStep).not.toHaveBeenCalled();
    });
  });
});
