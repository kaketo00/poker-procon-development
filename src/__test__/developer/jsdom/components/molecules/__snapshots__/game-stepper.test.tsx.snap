// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameStepper スナップショット/1ページ目 1`] = `
<DocumentFragment>
  <div
    class="css-hre5bs"
  >
    <button
      aria-label="Prev"
      class="chakra-button css-13s0th2"
      disabled=""
      type="button"
    >
      <span
        class="chakra-button__icon css-1wh2kri"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 16 16"
        >
          <path
            d="M11.354 1.646a.5.5 0 0 1 0 .708L5.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z"
            fill="currentColor"
          />
        </svg>
      </span>
      Prev
    </button>
    <div
      class="chakra-numberinput css-1j8xgar"
      value="1"
    >
      <input
        aria-label="pager"
        aria-valuemax="10"
        aria-valuemin="1"
        aria-valuenow="1"
        aria-valuetext="1"
        autocomplete="off"
        autocorrect="off"
        class="chakra-numberinput__field css-skouzv"
        inputmode="decimal"
        pattern="[0-9]*(.[0-9]+)?"
        role="spinbutton"
        type="text"
        value="1"
      />
      <div
        aria-hidden="true"
        class="css-1jj9yua"
      >
        <div
          class="css-1m5jnul"
          role="button"
          tabindex="-1"
        >
          <svg
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12.8,5.4c-0.377-0.504-1.223-0.504-1.6,0l-9,12c-0.228,0.303-0.264,0.708-0.095,1.047 C2.275,18.786,2.621,19,3,19h18c0.379,0,0.725-0.214,0.895-0.553c0.169-0.339,0.133-0.744-0.095-1.047L12.8,5.4z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          aria-disabled="true"
          class="css-1m5jnul"
          disabled=""
          role="button"
          tabindex="-1"
        >
          <svg
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M21,5H3C2.621,5,2.275,5.214,2.105,5.553C1.937,5.892,1.973,6.297,2.2,6.6l9,12 c0.188,0.252,0.485,0.4,0.8,0.4s0.611-0.148,0.8-0.4l9-12c0.228-0.303,0.264-0.708,0.095-1.047C21.725,5.214,21.379,5,21,5z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
    <p
      class="chakra-text css-paz7wr"
    >
      /
    </p>
    <p
      class="chakra-text css-paz7wr"
    >
      10
    </p>
    <button
      aria-label="Next"
      class="chakra-button css-13s0th2"
      type="button"
    >
      Next
      <span
        class="chakra-button__icon css-1hzyiq5"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 16 16"
        >
          <path
            d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
            fill="currentColor"
          />
        </svg>
      </span>
    </button>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`GameStepper スナップショット/最終ページ 1`] = `
<DocumentFragment>
  <div
    class="css-hre5bs"
  >
    <button
      aria-label="Prev"
      class="chakra-button css-13s0th2"
      type="button"
    >
      <span
        class="chakra-button__icon css-1wh2kri"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 16 16"
        >
          <path
            d="M11.354 1.646a.5.5 0 0 1 0 .708L5.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z"
            fill="currentColor"
          />
        </svg>
      </span>
      Prev
    </button>
    <div
      class="chakra-numberinput css-1j8xgar"
      value="10"
    >
      <input
        aria-label="pager"
        aria-valuemax="10"
        aria-valuemin="1"
        aria-valuenow="10"
        aria-valuetext="10"
        autocomplete="off"
        autocorrect="off"
        class="chakra-numberinput__field css-skouzv"
        inputmode="decimal"
        pattern="[0-9]*(.[0-9]+)?"
        role="spinbutton"
        type="text"
        value="10"
      />
      <div
        aria-hidden="true"
        class="css-1jj9yua"
      >
        <div
          aria-disabled="true"
          class="css-1m5jnul"
          disabled=""
          role="button"
          tabindex="-1"
        >
          <svg
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12.8,5.4c-0.377-0.504-1.223-0.504-1.6,0l-9,12c-0.228,0.303-0.264,0.708-0.095,1.047 C2.275,18.786,2.621,19,3,19h18c0.379,0,0.725-0.214,0.895-0.553c0.169-0.339,0.133-0.744-0.095-1.047L12.8,5.4z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="css-1m5jnul"
          role="button"
          tabindex="-1"
        >
          <svg
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M21,5H3C2.621,5,2.275,5.214,2.105,5.553C1.937,5.892,1.973,6.297,2.2,6.6l9,12 c0.188,0.252,0.485,0.4,0.8,0.4s0.611-0.148,0.8-0.4l9-12c0.228-0.303,0.264-0.708,0.095-1.047C21.725,5.214,21.379,5,21,5z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
    <p
      class="chakra-text css-paz7wr"
    >
      /
    </p>
    <p
      class="chakra-text css-paz7wr"
    >
      10
    </p>
    <button
      aria-label="Next"
      class="chakra-button css-13s0th2"
      disabled=""
      type="button"
    >
      Next
      <span
        class="chakra-button__icon css-1hzyiq5"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 16 16"
        >
          <path
            d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
            fill="currentColor"
          />
        </svg>
      </span>
    </button>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
